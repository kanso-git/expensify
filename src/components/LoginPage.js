import React from 'react';


export default class LoginPage extends React.Component{

    handleLogin =(e) =>{
      e.preventDefault();
      const username = e.target.username.value;
      const password = e.target.password.value;
      this.props.history.push('/dashboard');
    };


    render(){
      return (
          <div >
              <form onSubmit={this.handleLogin} >
                    username: <input type="text" id="username"/>
                    password: <input type="password" id="password"/>
                    <button > Login to Expensify </button>
              </form>
          </div>
      )

    }

}