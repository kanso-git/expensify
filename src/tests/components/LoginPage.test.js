import {shallow} from 'enzyme';
import LoginPage from '../../components/LoginPage';
import React from 'react';


test('should render login page ',()=>{

    const wrapper = shallow(<LoginPage/>);
    expect(wrapper).toMatchSnapshot();


})