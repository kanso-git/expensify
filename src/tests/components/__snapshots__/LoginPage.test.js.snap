// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page  1`] = `
<div>
  <form>
    username: 
    <input
      id="username"
      type="text"
    />
    password: 
    <input
      id="password"
      type="password"
    />
    <button
      id="submitBtn"
      onClick={[Function]}
      type="submit"
    >
       Login to Expensify 
    </button>
  </form>
</div>
`;
